proto/miner.proto

```diff
@@ -58,7 +58,7 @@ message MinerStartRequest{
 }
 
 message SocketAddr {
-    string addr = 1;
+    repeated string addrs = 1;
```

Comment:
It's not correct for a socket address to have multiple IPs. I'd rather return multiple SocketAddr instead.